src/urllib3/response.py

```diff
@@ -449,19 +469,21 @@ def drain_conn(self) -> None:
             pass
 
     @property
-    def data(self):
+    def data(self) -> Optional[Union[bytes, str]]:
```

Comment:
Should this be `-> bytes`?

Reply:
we have `return self._body` in this function
https://github.com/urllib3/urllib3/blob/main/src/urllib3/response.py#L455

And `_body` can be `Union[str, bytes]` https://github.com/urllib3/urllib3/blob/main/src/urllib3/response.py#L414-L415
Also, The function doesn't have a return(I added a `return None`) 
So, I changed the return type to `Optional[Union[bytes, str]]`

Reply:
after changing the return type to `bytes`, I got:
```
src/urllib3/response.py:475: error: Incompatible return value type (got "Union[str, bytes]", expected "bytes")
src/urllib3/response.py:478: error: Incompatible return value type (got "Optional[bytes]", expected "bytes")
src/urllib3/response.py:480: error: Incompatible return value type (got "None", expected "bytes")
```

We have to ignore them for now.